### DISCUSSION TITLE
UTC Timestamp column for jsPsych

### DISCUSSION THREAD

**Discussion by psiwex** [jspsych/jsPsych#3195] [Q&A]
I am using a flanker task with an external EEG device that timestamps in UTC. I am looking to add a new column to the output file that has the UTC timestamp from the local machine, where the UTC value for each event is entered. How might I do that? Thanks!

**Comment by jodeleeuw** [comment]
If the goal is to timestamp the flanker display on the jsPsych side you can use the `on_load` event to grab the current `Date.now()` value and then write it to the data in the `on_finish` event.

```js
const trial = {
	// some other stuff here to display flanker
	on_load: () => {
		load_event_utc = new Date(Date.now()).toUTCString();
	},
	on_finish: (data) => {
		data.utc_timestamp = load_event_utc
	}
}
```

For EEG, just be aware that this won't give you a super precise onset of the flanker stimulus. With JavaScript there is currently no way to directly measure the moment that the screen updates. Instead, what this is doing is measuring the moment that the code to update the display finishes running. The actual display update could happen a few milliseconds before or after this point, depending on where in the refresh cycle the event loop is.

**Reply by psiwex** [reply]
Thanks. Has anyone ever used jsPsych for the Emotion Recognition Task or Emotional Faces Assessment Task (EFAT)? Thanks.
